core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -1436,6 +1440,38 @@ protected Object visitRow(Row node, Object context)
             return rowType.getObject(blockBuilder, 0);
         }
 
+        @Override
+        protected Object visitFormat(Format node, Object context)
+        {
+            Object format = processWithExceptionHandling(node.getArguments().get(0), context);
+            if (format == null) {
+                return null;
+            }
+
+            // FORMAT(a, b, c, d, ...) gets desugared into $format(a, row(b, c, d, ...))
+            List<Expression> arguments = node.getArguments().subList(1, node.getArguments().size());
+            List<Type> argumentTypes = arguments.stream()
+                    .map(this::type)
+                    .collect(toImmutableList());
+
+            RowType rowType = anonymous(argumentTypes);
+            ResolvedFunction function = plannerContext.getMetadata()
+                    .resolveFunction(session, QualifiedName.of(FormatFunction.NAME), TypeSignatureProvider.fromTypes(VARCHAR, rowType));
+
+            // Construct a row with arguments [1..n] and invoke the underlying function
+            BlockBuilder rowBuilder = new RowBlockBuilder(argumentTypes, null, 1);
+            BlockBuilder singleRowBlockWriter = rowBuilder.beginBlockEntry();
+            for (int i = 0; i < arguments.size(); ++i) {
+                writeNativeValue(argumentTypes.get(i), singleRowBlockWriter, processWithExceptionHandling(arguments.get(i), context));
```

Comment:
Please note that the processed format or any of the arguments might be unresolved. In such case, we should return a (desugared) function call instead of invoking a function.